File: Code/Chapter 4 Primitive Obsession/4.02 Reifying Primitive Data/smallobjects.js
// Samples borrowed with permission from
// https://towardsdev.com/why-a-host-is-not-a-string-and-a-port-is-not-an-integer-595c182d817c

const server = Port.parse(this, "www.kivakit.org:8080");
// Port is a smallobject with responsibilities and protocol

let in = port.open(this);
const uri = port.asUri(this);
const address = port.asInetSocketAddress();
const path = port.path(this, "/index.html");

File: Code/Chapter 04 Primitive Obsession/4.02 Reifying Primitive Data/smallobjects.java
Port server = Port.parse(this, "www.example.org:8080");
// Port is a smallobject with responsibilities and protocol

Port in = server.open(this); // returns a port, not a number
URI uri = server.asUri(this); // returns an URI
InetSocketAddress address = server.asInetSocketAddress();
// returns an Address
Path path = server.path(this, "/index.html"); // returns a Path
// all of them are validated small bijection objects
// with very few and precise
// responsibilities